src/Catalyst.Core.Modules.Kvm/DeltaExecutor.cs

```diff
@@ -0,0 +1,411 @@
+            // {
+            //     /* discarding the other branch data - chain reorganization */
+            //     Metrics.Reorganizations++;
+            //     _storageProvider.Reset();
+            //     _stateProvider.Reset();
+            //     _stateProvider.StateRoot = branchStateRoot;
+            // }
+            
+            foreach (PublicEntry publicEntry in delta.PublicEntries)
+            {
+                ContractEntry contractEntry = new ContractEntry();
+                contractEntry.Base = publicEntry.Base;
+                contractEntry.Amount = publicEntry.Amount;
+                contractEntry.Data = ByteString.Empty;
+                contractEntry.GasLimit = 21000;
+                Execute(contractEntry, delta, txTracer, readOnly, recipientOverride);
+            }
+            
+            // revert state if any fails (take snapshot)
+            foreach (ContractEntry contractEntry in delta.ContractEntries)
+            {
+                Execute(contractEntry, delta, txTracer, readOnly, recipientOverride);
+            }
+
+            if (!readOnly)
+            {
+                // we should assign block rewards here (or in Ledger)
+                _stateProvider.CommitTree();
+                _storageProvider.CommitTrees();
+            }
+
+            // if ((options & ProcessingOptions.ReadOnlyChain) != 0)
+            // {
+            //     Restore(stateSnapshot, codeSnapshot, snapshotStateRoot);
+            // }
+            // else
+            // {
+            //     _stateDb.Commit();
+            //     _codeDb.Commit();
+            // }
+        }
+        
+        private void Execute(ContractEntry entry, Delta delta, ITxTracer txTracer, bool readOnly, Address recipientOverride = null)
+        {
+            var stateUpdate = ToStateUpdate(delta);
+            long gasLimit = entry.GasLimit;
+            UInt256 gasPrice = entry.GasPrice;
+            Address recipient = recipientOverride ?? GetAccountAddress(entry.Base.ReceiverPublicKey);
+            Address sender = GetAccountAddress(entry.Base.SenderPublicKey);
+            var value = entry.Amount.ToUInt256();
+
+            IReleaseSpec spec = _specProvider.GetSpec(stateUpdate.Number);
+            byte[] machineCode = entry.IsValidDeploymentEntry ? entry.Data.ToByteArray() : null;
+            byte[] data = entry.Data.ToByteArray();
+
+            // if (_logger.IsTrace) _logger.Trace($"Executing tx {transaction.Hash}");
+            if (sender == null)
+            {
+                // TraceLogInvalidTx(entry, "SENDER_NOT_SPECIFIED");
+                QuickFail(entry, stateUpdate, txTracer, readOnly);
+                return;
+            }
+            
+            long intrinsicGas = CalculateIntrinsicGas(entry, spec);
+            
+            //if (_logger.IsTrace) _logger.Trace($"Intrinsic gas calculated for {entry}: " + intrinsicGas);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/app/catalyst-network/Catalyst.Node/pullRequest?prid=4350015)